### PR TITLE
tsweb: replace NewMux with a more flexible DebugHandler.

### PR DESCRIPTION
**Pull request by danderson**
Signed-off-by: David Anderson <danderson@tailscale.com>

### CODE REVIEW

File: tsweb/debug.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+func (d *DebugHandler) KV(k string, v func() interface{}) {
+	d.kvs = append(d.kvs, func(w io.Writer) { fmt.Fprintf(w, "<li><b>%s:</b> %v</li>", k, v()) })
+}
+
+// URL adds a URL and description list item to /debug/.
+func (d *DebugHandler) URL(url, desc string) {
+	if desc != "" {
+		desc = " (" + desc + ")"
+	}
+	d.urls = append(d.urls, fmt.Sprintf(`<li><a href="%s">%s</a>%s</li>`, url, url, desc))
```
Comment (bradfitz) [inline tsweb/debug.go:96]: html.EscapeString(desc) ?

File: tsweb/debug.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	"tailscale.com/version"
+)
+
+// DebugHandler is an http.Handler that serves a debugging "homepage",
+// and provides helpers to register more debug endpoints nad
+// reporting.
+type DebugHandler struct {
+	mux      *http.ServeMux
+	kvs      []func(io.Writer)
+	urls     []string
```
Comment (bradfitz) [inline tsweb/debug.go:26]: document what this string is

File: tsweb/debug.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+// entry in /debug/ for it.
+func (d *DebugHandler) Handle(slug, desc string, handler http.Handler) {
+	href := "/debug/" + slug
+	d.mux.Handle(href, Protected(handler))
+	d.URL(href, desc)
+}
+
+// KV adds a key/value list item to /debug/ that prints the value of v().
+func (d *DebugHandler) KV(k string, v func() interface{}) {
+	d.kvs = append(d.kvs, func(w io.Writer) { fmt.Fprintf(w, "<li><b>%s:</b> %v</li>", k, v()) })
```
Comment (bradfitz) [inline tsweb/debug.go:88]: what can the v func return? No HTML escaping?

File: tsweb/debug.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+// URL adds a URL and description list item to /debug/.
```
Comment (bradfitz) [inline tsweb/debug.go:91]: what is the "url" value? like a path? a full URL? give some example values in the docs.

File: tsweb/debug.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+	"tailscale.com/version"
+)
+
+// DebugHandler is an http.Handler that serves a debugging "homepage",
+// and provides helpers to register more debug endpoints nad
+// reporting.
+type DebugHandler struct {
+	mux      *http.ServeMux
+	kvs      []func(io.Writer)
```
Comment (bradfitz) [inline tsweb/debug.go:25]: docs on all these?

File: tsweb/debug.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+// creating it if necessary.
+func Debugger(mux *http.ServeMux) *DebugHandler {
+	h, pat := mux.Handler(&http.Request{URL: &url.URL{Path: "/debug/"}})
+	if d, ok := h.(*DebugHandler); ok && pat == "/debug/" {
+		return d
+	}
+	ret := &DebugHandler{
+		mux: mux,
+	}
+	mux.Handle("/debug/", Protected(ret))
```
Comment (bradfitz) [inline tsweb/debug.go:40]: add a test for this that the line 33 lookup actually works and gets a == equal DebugHandler on 2nd call

File: tsweb/debug.go (hunk 4 above)
Comment (danderson) [reply tsweb/debug.go:115]: added example tests with value values.

**Pull request merged**